kivymd/uix/button/button.py

```diff
@@ -852,16 +819,47 @@ class ButtonContentsIcon():
     :attr:`icon_color` is an :class:`~kivy.properties.ColorProperty`
     and defaults to `None`.
     """
+
+    theme_icon_color = OptionProperty(None, options=THEME_TEXT_COLOR_OPTIONS)
+    """
+    Button icon type. Available options are: (`"Primary"`, `"Secondary"`,
+    `"Hint"`, `"Error"`, `"Custom"`, `"ContrastParentBackground"`).
+
+    :attr:`theme_icon_color` is an :class:`~kivy.properties.OptionProperty`
+    and defaults to `None` (depends on button class).
+    """
     _icon_color = ColorProperty(None)
+    _theme_icon_color = OptionProperty(None, options=THEME_TEXT_COLOR_OPTIONS)
+    _default_icon_color = ColorProperty(None)
+    _default_theme_icon_color = StringProperty('Primary')
 
-    def set_text_color(self, *args):
-        super().set_text_color()
-        self._icon_color = self.icon_color or self._text_color
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.bind(theme_icon_color=self.set_text_color)
 
+    def set_text_color(self, *args) -> NoReturn:
+        """Set _theme_icon_color and _icon_color based on defaults
+        and options.
+        """
+        super().set_text_color(*args)
```

Comment:
Indent one docstring in methods.

Reply:
ok (and will in all other cases as well).